src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/BackgroundSink.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Serilog.Core;
+using Serilog.Events;
+
+// TODO: Might want to consider using https://github.com/jezzsantos/Serilog.Sinks.Async 
+// instead of this, once that supports netstandard
+namespace Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Internal
+{
+    /// <summary>
+    /// A background sink for Serilog.
+    /// </summary>
+    public class BackgroundSink : ILogEventSink, IDisposable
```

Comment:
We have this convention that anything in the `Internal` namespace can be public but it's still internal (so we can change it). We do that to avoid InternalsVisibleTo and still be able to test internal code